pkg/kusionctl/cmd/compile/options.go

```diff
@@ -94,8 +94,11 @@ func (o *CompileOptions) PreSet(preCheck func(cur string) bool) {
 		return
 	}
 
-	if len(o.Settings) == 0 && o.Output == "" && len(o.Filenames) == 0 {
+	if len(o.Settings) == 0 {
```

Comment:
1. should settings be set up to default value when the `o.Filenames` is not empty?
since the settings file may configs entrance file list, which are conflict with `o.Filenames`

2. should settings be set up to default value when the `o.Output` is not empty?
since the settings file may configs output file path, which is conflict with `o.Output`

Reply:
the `settings.yaml` file can define entrance files, so do the cmd args, the o.Filenames is read from cmd args, not a flag.  and `settings.yaml` can also do the output file path ? really?

Reply:
about settings.yaml, you can see https://kusionstack.io/docs/reference/lang/lang/tour/#arguments-with-setting-files and checkout the kcl_cli_configs key

Reply:
`kcl_cli_configs` is used in `kcl.yaml`, `kcl_options` is used in `settings.yaml`

Reply:
> `kcl_cli_configs` is used in `kcl.yaml`, `kcl_options` is used in `settings.yaml` for now

kcl.yaml and ci-test/settings.yaml are all settings files, only distinguish in names

Reply:
cc @Peefy

Reply:
accoding to @Peefy, user-specified filenames will override filenames in `kcl.yaml`, so it's is ok to compile